client/client-core/src/main/java/software/amazon/smithy/java/client/core/Client.java

```diff
@@ -75,20 +76,26 @@ protected <I extends SerializableStruct, O extends SerializableStruct> Completab
             ApiOperation<I, O> operation,
             RequestOverrideConfig overrideConfig
     ) {
-        // Create a copy of the type registry that adds the errors this operation can encounter.
-        TypeRegistry operationRegistry = TypeRegistry.compose(operation.errorRegistry(), typeRegistry);
-
-        ClientPipeline<?, ?> callPipeline;
-        ClientInterceptor callInterceptor;
-        IdentityResolvers callIdentityResolvers;
-        ClientConfig callConfig;
-        if (overrideConfig == null) {
-            callConfig = config;
-            callPipeline = pipeline;
-            callInterceptor = interceptor;
-            callIdentityResolvers = identityResolvers;
-        } else {
-            callConfig = config.withRequestOverride(overrideConfig);
+        ClientPipeline<?, ?> callPipeline = pipeline;
+        IdentityResolvers callIdentityResolvers = identityResolvers;
+        ClientConfig callConfig = config;
+        ClientInterceptor callInterceptor = interceptor;
+
+        // Apply the given override before potentially applying interceptor based overrides.
+        var needsRebuild = false;
+        if (overrideConfig != null) {
+            needsRebuild = true;
+            callConfig = callConfig.withRequestOverride(overrideConfig);
+            callInterceptor = ClientInterceptor.chain(callConfig.interceptors());
+        }
+
+        var updatedConfig = callInterceptor.modifyBeforeCall(new CallHook<>(operation, callConfig, input));
```

Comment:
I think this should be applied before request overrides. We should let the most specific overrides dominate the configuration, and client-level settings are less specific than settings being applied on individual calls. I know I would be confused if overrides I passed to `Client::call` were silently ignored or overwritten.